### ISSUE TITLE
Allow creating data_item connections from API

### ISSUE THREAD

**Issue by Brow71189** [nion-software/nionswift#246]
It would be great to have a way to specify connections between data_items from plugins and scripts, so that you get the little thumbnails that you get from computations.

**Comment by cmeyer** [comment]
@Brow71189 Can you give your specific use case for this?

**Comment by Brow71189** [comment]
This is 6 years old, so I cannot remember the exact application I had in mind back then. But I still think it would be nice if you could show relations between data items that  you create by a script or a plug-in.

**Comment by cmeyer** [comment]
Yes, I agree - I just want to make a specific use case. The ones I know about are: computations; composite line plots and future composite raster images; and items related by acquisition, which will be covered by acquisition objects. Are there others you can think of?

Alternatively, what kinds of API would this be? 'add_relationship_between_items(relationship_type, items)' ?
